examples/parallel_profiling.py

```diff
@@ -0,0 +1,144 @@
+"""
+================================================
+Memory and CPU consumption using joblib.Parallel
+================================================
+
+This example illustrates some important aspects to keep in mind when using
+:class:`joblib.Parallel`. We will focus on different patterns resulting to the
+same outcomes but which trigger different internal behaviors. Those differences
+will impact the memory consumption and computation time.
+
+To illustrate those differences, the moving average from a large array will be
+computed. Therefore, the average can be computed in parallel at different
+location of the signal.
+
+"""
+from __future__ import division
+
+import os
+import time
+import gc
+import numpy as np
+from memory_profiler import memory_usage
+
+import joblib
+from joblib import Parallel, delayed
+
+print(__doc__)
+
+###############################################################################
+# `memory_used` and `print_info` are two helper functions being used
+# extensively later on. The former function allows to track the memory
+# consumption during the call of a function while the latter output profiling
+# information.
+
+
+def memory_used(func, *args, **kwargs):
+    gc.collect()
+    mem_use = memory_usage((func, args, kwargs), interval=.00001,
+                           multiprocess=True, include_children=True)
+    return max(mem_use) - min(mem_use)
+
+
+def print_info(mem_used_avg, elapsed_time):
+    print('-' * 79)
```

Comment:
You don't need this for things to look nice on the HTML output. I think that it can be removed.